### PR TITLE
feat(modal): Expose a closable property to be able to hide x icon

### PR DESCRIPTION
**Pull request by mistermalm**
## **Describe pull-request**  
Expose a collapsible property to be able to hide x icon

## **Issue Linking:**  
- **Jira:** [CDEP-3294](https://tegel.atlassian.net/browse/CDEP-3294)    

## **How to test**  
1. Set closable property to false.
2. See that the close button is no more.

## **Checklist before submission**
- [ ] I have added unit tests for my changes (if applicable)
- [x] All existing tests pass
- [] I have updated the documentation (if applicable)

## **Suggested test steps**
- [ ] Browser testing (Chrome, Safari, Firefox) 
- [ ] Keyboard operability
- [ ] Interactive elements have labels.
- [x] Storybook controls
- [x] Design/controls/props is aligned with other components 
- [ ] Dark/light mode and variants 
- [ ] Input fields – values should be displayed properly 
- [ ] Events

## **Additional context**  
Unit tests will be added if requested.

[CDEP-3294]: https://tegel.atlassian.net/browse/CDEP-3294?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

**Review by timrombergjakobsson — COMMENTED** [review]
Nice man! Maybe we should add a test case for covering this also?

**Review by mJarsater — COMMENTED** [review]
I think this has been requested before and if I remember correctly this is a design decision. That the modal should  **always** have a close button?

**Comment by theJohnnyMe** [comment]
> I think this has been requested before and if I remember correctly this is a design decision. That the modal should **always** have a close button?

I had the same memory so I asked yesterday about it but designers approved it. I am not sure if was I confusing it with the banner component thou.

**Comment by timrombergjakobsson** [comment]
> > I think this has been requested before and if I remember correctly this is a design decision. That the modal should **always** have a close button?
> 
> I had the same memory so I asked yesterday about it but designers approved it. I am not sure if was I confusing it with the banner component thou.

Yes, I think memory serves you correct. Although as I believe you can hide it in Figma, it made sense to have this flexibility in the component also. But please, @charlottelorensson or Laurens, you could also give your view on it.

**Review by nathalielindqvist — COMMENTED** [review]
Works as expected, nicely done! Maybe introduce a test case for it to cover all bases? 😄

**Comment by mistermalm** [comment]
I agree that it's not best practice UX to disable the close button, sometimes users do not want to make a decision, even if it has a cancel button, they just want to close.

Will add tests as well to ensure quality

File: packages/core/src/components/modal/modal.stories.tsx [hunk 1]
```diff
@@ -72,6 +72,13 @@ export default {
         type: 'boolean',
       },
     },
+    closable: {
+      name: 'Closable',
+      description: 'Controls visibility of the close button.',
+      control: {
+        type: 'boolean',
+      },
```
Comment (theJohnnyMe) [inline packages/core/src/components/modal/modal.stories.tsx:80]: Add here that defaultValue is "true" so it reflects defaultValue from modal.tsx

File: packages/core/src/components/modal/modal.tsx [hunk 2]
```diff
@@ -46,6 +46,9 @@ export class TdsModal {
    * will be decided by this prop and will need to be controlled from the outside. */
   @Prop() show: boolean;
 
+  /** Enables the close button. */
```
Comment (theJohnnyMe) [inline packages/core/src/components/modal/modal.tsx:49]:
```suggestion
  /** Shows or hides the close [X] button. */
```
Just to be more specific. Enable might mean that button is there but we do something with preventDefault etc...

**Comment by mistermalm** [comment]
I have now pushed tests for close button aswell

**Review by nathalielindqvist — APPROVED** [review]
LGTM, great work! 🚀

**Review by theJohnnyMe — APPROVED** [review]
LGTM

**Pull request merged**